src/features/transaction-wizard/components/transaction-builder-form.tsx

```diff
@@ -0,0 +1,153 @@
+import { asTransaction } from '@/features/transactions/mappers'
+import { assetSummaryResolver } from '@/features/assets/data'
+import { abiMethodResolver } from '@/features/abi-methods/data'
+import { getIndexerTransactionFromAlgodTransaction } from '@algorandfoundation/algokit-subscriber/transform'
+import { TransactionsGraph, TransactionsGraphData } from '@/features/transactions-graph'
+import { TransactionLink } from '@/features/transactions/components/transaction-link'
+import { cn } from '@/features/common/utils'
+import { DescriptionList } from '@/features/common/components/description-list'
+import { transactionIdLabel } from '@/features/transactions/components/transaction-info'
+
+type Props<TSchema extends z.ZodSchema> = {
+  buildableTransaction: BuildableTransaction<TSchema>
+  defaultSender?: string
+}
+
+const connectWalletMessage = 'Please connect a wallet'
+
+type SendTransactionResult = {
+  transactionId: string
+  transactionsGraphData: TransactionsGraphData
+}
+
+export function TransactionBuilderForm<TSchema extends z.ZodSchema>({ buildableTransaction, defaultSender }: Props<TSchema>) {
+  const { activeAddress, signer } = useWallet()
+
+  const [sendTransactionResult, setSendTransactionResult] = useState<SendTransactionResult | undefined>(undefined)
+
+  const sendTransaction = useCallback(
+    async (values: Parameters<typeof buildableTransaction.createTransaction>[0]) => {
+      if (!activeAddress) {
+        throw new Error(connectWalletMessage)
+      }
+
+      const transaction = await buildableTransaction.createTransaction(values)
+
+      // This isn't amazing, we should add support to the util-ts AlgorandClient for sending a single transaction.
+      const atc = new algosdk.AtomicTransactionComposer()
+      atc.addTransaction({ txn: transaction, signer })
+      const result = await algorandClient.newGroup().addAtc(atc).execute()
+
+      const transactionId = result.txIds[0]
+      const confirmation = result.confirmations[0]
+      const transactionResult = getIndexerTransactionFromAlgodTransaction({
+        blockTransaction: {
+          txn: confirmation.txn.txn,
+        },
+        roundOffset: 0,
+        roundIndex: 0,
+        genesisHash: confirmation.txn.txn.gh,
+        genesisId: confirmation.txn.txn.gen,
+        roundNumber: Number(confirmation.confirmedRound),
+        roundTimestamp: Math.floor(Date.now() / 1000),
+        transaction: result.transactions[0],
+      })
+      const transactionsGraphData = asTransactionsGraphData([asTransaction(transactionResult, assetSummaryResolver, abiMethodResolver)])
+
+      setSendTransactionResult({
+        transactionId,
+        transactionsGraphData,
+      })
+
+      toast.success('Transaction sent successfully')
+    },
+    [activeAddress, buildableTransaction, signer]
+  )
+
+  // NOTE: Number fields need the default value `'' as unknown as undefined` in order to be cleared.
```

Comment:
Nice fix, I remember it now. This was the reason why we had form validation errors on the numbers before.

Reply:
It works, however it's a bit annoying having to do this. It also need to be coupled with the `numberSchema` function I created in order to work properly, so there are definitely a few traps/sharp edges.